src/aleph_vrf/coordinator/vrf.py

```diff
@@ -80,12 +90,20 @@ async def _generate_vrf(
 
     nonce = generate_nonce()
 
+    if request_id:
+        existing_message = await get_existing_vrf_message(aleph_client, request_id)
+        if existing_message:
+            return PublishedVRFResponse.from_vrf_post_message(existing_message)
+
+    if not request_id:
+        request_id = str(uuid4())
+
     vrf_request = VRFRequest(
         nb_bytes=nb_bytes,
         nb_executors=nb_executors,
         nonce=nonce,
         vrf_function=vrf_function,
-        request_id=RequestId(str(uuid4())),
+        request_id=RequestId(request_id),
```

Comment:
Would it not be clearer to use an `or` and remove the `if not request_id` above ?

```suggestion
        request_id=RequestId(request_id or str(uuid4()),
```